gradle/release.gradle

```diff
@@ -30,7 +38,43 @@ task sourceJar(type: Jar) {
     classifier "sources"
 }
 
+task validatePublishOptions() {
+    description = "Ensures assembleRelease is run with the correct NDK library version and ABI list"
+    doFirst {
+        def ndkVersion
+        try {
+            def props = new Properties()
+            file(System.getenv('ANDROID_NDK_HOME') + "/source.properties").withInputStream {
+                props.load(it)
+            }
+            ndkVersion = props.getProperty("Pkg.Revision")
+        } catch (Exception e) {
+            ndkVersion = "UNKNOWN"
+        }
+
+        if (!ndkVersion.startsWith("${expectedNdkVersion}.")) {
+            throw new GradleException("${project.name} must be released with ${expectedNdkVersion}.x of the NDK - found ${ndkVersion}")
+        }
+
+        if (project.android.defaultConfig.ndk.abiFilters != null) {
+            for (abiList in expectedAbiLists) {
+                if (abiList == project.android.defaultConfig.ndk.abiFilters as Set) {
+                    return
+                }
+            }
+            throw new GradleException("Unexpected list of ABIs for release: ${project.android.defaultConfig.ndk.abiFilters}")
+        }
+    }
+}
+
+tasks.whenTaskAdded { task ->
+    if (task.name == 'assembleRelease') {
```

Comment:
It might be worth depending on the 'publish' and 'bintrayUpload' tasks as these can be invoked without having run `assembleRelease`.